lib/update-fetch.js

```diff
@@ -95,58 +99,68 @@ function loadLocalBinary(path) {
   });
 }
 
+function fetchRestoreFiles() {
+  var fileMap = {
+    uboot: RESTORE_UBOOT_FILE,
+    squashfs: RESTORE_SQASHFS_FILE
+  };
+  return downloadTgz(RESTORE_TGZ_URL, fileMap);
+}
+
 /*
   Accepts a build name and attempts to fetch
   the build images from the server. Returns build contents
   in a Promise
 */
 function fetchBuild(build) {
+  var tgzUrl = urljoin(FIRMWARE_PATH, build.sha + '.tar.gz');
+  var fileMap = {
+    firmware: FIRMWARE_BINARY_FILE,
+    openwrt: OPENWRT_BINARY_FILE
+  };
+  return downloadTgz(tgzUrl, fileMap);
+}
+
+function downloadTgz(tgzUrl, fileMap) {
   return new Promise(function(resolve, reject) {
-    logs.info('Beginning update download. This could take a couple minutes..');
+    logs.info('Beginning download. This could take a couple minutes..');
 
-    var binaries = {
-      firmware: undefined,
-      openwrt: undefined,
-    };
+    var files = {};
 
-    // Fetch the list of available builds
+    // Fetch the list of available files
     extract.on('entry', function(header, stream, callback) {
       // The buffer to save incoming data to
       // The filename of this entry
-      var filename = path.basename(header.name);
-      // This entry is the openwrt binary
-      if (filename === OPENWRT_BINARY_FILE) {
-        // Save incoming data to the openwrt buffer
-        streamToBuffer(stream, function(err, buffer) {
-          binaries.openwrt = buffer;
-          callback();
-        });
-      }
-      // This entry is the firmware binary
-      else if (filename === FIRMWARE_BINARY_FILE) {
-        // Save incoming data to the firmware buffer
-        streamToBuffer(stream, function(err, buffer) {
-          binaries.firmware = buffer;
-          callback();
-        });
-      } else {
-        callback();
+      var tgzFilename = path.basename(header.name);
+
+      for (var key in fileMap) {
+        var expectedFilename = fileMap[key];
+        if (tgzFilename === expectedFilename) {
+          return streamToBuffer(stream, function(err, buffer) {
+            files[key] = buffer;
+            callback();
+          });
+        }
       }
+      callback();
     });
 
     extract.once('finish', function() {
-      if (!binaries.firmware.length || !binaries.openwrt.length) {
-        return reject(new Error('Fetched binary wasn\'t formatted properly.'));
-      } else {
-        logs.info('Download complete!');
-        return resolve(binaries);
+      for (var key in files) {
+        var file = files[key];
+        if (!file.length) {
+          return reject(new Error('Fetched file wasn\'t formatted properly.'));
```

Comment:
@HipsterBrown someone with AWS permissions would need to push a properly formatted build up. Perhaps we can just suggest they create an issue so we're aware there's a problem?